### PR TITLE
Command clicking should "xor" selection

### LINKED ISSUE #272: Command clicking should "xor" selection
If you have command key held down and click on something, it should add it to the selection it if it wasn't selected or unselect it if it was already selected.

### PR DESCRIPTION
**Pull request by nanot1m**
Fixes #272

### CODE REVIEW

**Comment by dwelle** [comment]
Cool. One note: as per the issue, `Cmd` shouldn't play a role in this (we decided that only `Shift` should).

**Comment by nanot1m** [comment]
Removed Cmd key handling

File: src/index.tsx [hunk 1]
```diff
@@ -544,6 +544,9 @@ export class App extends React.Component<{}, AppState> {
             let resizeHandle: ResizeTestType = false;
             let isDraggingElements = false;
             let isResizingElements = false;
+            let someElementIsDragged = false;
```
Comment (dwelle) [inline src/index.tsx]: Not clear what's the difference between `someElementIsDragged` & `isDraggingElements`

Comment (nanot1m) [reply src/index.tsx]: `someElementIsDragged` changes to true if some dragging occured. `isDraggingElements` changes to true if we click to already selected element

Comment (dwelle) [reply src/index.tsx]: Maybe there's some conceptual difference, but I don't see any difference in code logic. `someElementIsDragged` is set unconditionally on `isDraggingElements`.

Even if we decided to distinguish between the two (but I still don't see any point now), we should at least rename those variables to reflect it. As it stands those names are identical, which adds to the confusion.

Comment (nanot1m) [reply src/index.tsx]: Key difference if that `someElementIsDragged` is set on **mousemove** event, when `isDraggingElements` is set on **onmousedown** event.

On the `mouseup` event, we need to ensure that there was no dragging... so the variable `someElementIsDragged` was added to keep that information.

Maybe we can rename them as `isDraggingElements` -> `readyToDrag` and `someElementIsDragged ` -> `dragOccured`?

**Comment by dwelle** [comment]
Alright, seems ok. Merging.

Btw, in the future, can you please create branches in this repo? That way it's easier to get checkout the code for when one wants to review from editor (you don't need to clone a repo and install deps all over again).

**Pull request merged**